src/drivers/drv_sensor.h

```diff
@@ -106,7 +106,7 @@
 #define DRV_BARO_DEVTYPE_LPS22HB	0x58
 #define DRV_ACC_DEVTYPE_ADIS16477	0x59
 #define DRV_GYR_DEVTYPE_ADIS16477	0x60
-
+#define DRV_BATT_DEVTYPE_BQ40Z50	0x61
```

Comment:
Is this needed?

Reply:
Probably not, I included it because I the driver I was using as a template (baro: LPS22HB) had included the device class information in the constructor:

	// set the device type from the interface
	_device_id.devid_s.bus_type = _interface->get_device_bus_type();
	_device_id.devid_s.bus = _interface->get_device_bus();
	_device_id.devid_s.address = _interface->get_device_address();
	_device_id.devid_s.devtype = DRV_BARO_DEVTYPE_LPS22HB;

I am not sure the intended usage of this information. I am still learning PX4 architecture so if I am doing anything that seems non-standard or plain wrong please let me know! I would love any suggestions about the best way to write a generic driver class.

Reply:
Device ids are identification used for storing calibration parameters. They're exposed over uORB (eg sensor_accel, sensor_gyro, etc) and the IOCTL interface (/dev/accel0, /dev/gyro0, etc).

I don't see a need for one with the current battery/system power architecture, but that may change in the future.